libraries/lib-track/Track.h

```diff
@@ -399,7 +399,8 @@ class TRACK_API Track /* not final */
    //! Subclass responsibility implements only a part of Duplicate(), copying
    //! the track data proper (not associated data such as for groups and views)
    /*!
-    @param unstretchInterval If set, this time interval's stretching must be applied.
+    @param unstretchInterval If set and of positive duration, this time
```

Comment:
I don't think someone should be able to call `ApplyStretchRatio` with an empty interval. I'd rather have `@pre `!unstretchInterval || unstretchInterval->second > unstretchInterval->first`. Then, in EffectOutputTracks.cpp, if the interval is empty, call `WaveTrack::Duplicate` with an `std::nullopt` (or equivalent).

Reply:
I realize that the `unstretchInterval` arguments of `Track::Duplicate` and `WaveTrack::ApplyStretchRatio` have opposite meanings: stretch nothing in the former and everything in the latter. Maybe we should remove that argument from `Duplicate` and `Clone` and let the callers make a call after that to `ApplyStretchRatio`.